go-controller/pkg/ovn/multihoming_test.go

```diff
@@ -210,6 +210,11 @@ func (em *secondaryNetworkExpectationMachine) expectedLogicalSwitchesAndPorts()
 					"subnet":      subnet.CIDR.String(),
 				}
 			}
+
+			// TODO: why don't L2 switches have otherConfig set ?... Fishy to say the least ...
```

Comment:
did you figure this out? maybe in a later commit I haven't reached yet. But I see in my KIND deployment there is otherConfig for a layer 2 topology switch:
```
[root@ovn-control-plane ~]# ovn-nbctl list logical_switch  default.l2.primary_ovn_layer2_switch
_uuid               : e60d90fb-d46b-4dc6-b01d-87e1621847e7
acls                : [5891d5f2-66d9-482c-941d-606ecd924e64]
copp                : []
dns_records         : []
external_ids        : {"k8s.ovn.org/network"=default.l2-primary, "k8s.ovn.org/topology"=layer2}
forwarding_groups   : []
load_balancer       : []
load_balancer_group : []
name                : default.l2.primary_ovn_layer2_switch
other_config        : {interconn-ts=default.l2.primary_ovn_layer2_switch, mcast_flood_unregistered="true", mcast_querier="false", mcast_snoop="true", requested-tnl-key="16711685", subnet="10.20.0.0/16"}
ports               : [636369ac-8245-4c9f-899c-a004bf2b366b, aafc09a5-2abe-4d73-a6b9-922017b51864]
```

at least I would expect some conditional like below for if its IC...

Reply:
OK, I get it now. 

This only happens for unit tests. Our unit tests are **not** actually invoking the layer2 secondary network controller init which actually creates the switch. Thus, I suspect the unit test framework actually creates a minimal version of the switch so it can attach ports to. 

I could change the test to actually invoke this function, which would mostly make closer to reality, but then, I'd need to provision A LOT more objects (COPP & meters at least).

This is the diff I get when I tried it out (extra OVN entities that are provisioned):
```
[
            <*nbdb.Meter | 0xc00114d200>{
                UUID: "32e3fa18-0a26-4cec-9fdd-54b399f94c9d",
                Bands: [
                    "2e7bd3d6-1ba2-475e-af2e-db8da42551be",
                ],
                ExternalIDs: nil,
                Fair: true,
                Name: "bfd-rate-limiter",
                Unit: "pktps",
            },
            <*nbdb.Meter | 0xc00114d320>{
                UUID: "32e96240-7ae8-44f7-bfbb-5655f280c623",
                Bands: [
                    "2e7bd3d6-1ba2-475e-af2e-db8da42551be",
                ],
                ExternalIDs: nil,
                Fair: true,
                Name: "svc-monitor-rate-limiter",
                Unit: "pktps",
            },
            <*nbdb.Meter | 0xc00114d380>{
                UUID: "2251e0ef-aeb1-4671-807b-c839b264f3b0",
                Bands: [
                    "2e7bd3d6-1ba2-475e-af2e-db8da42551be",
                ],
                ExternalIDs: nil,
                Fair: true,
                Name: "reject-rate-limiter",
                Unit: "pktps",
            },
            <*nbdb.Meter | 0xc00114d3e0>{
                UUID: "39c3d5b0-3967-466c-a700-f8f68f1ff186",
                Bands: [
                    "2e7bd3d6-1ba2-475e-af2e-db8da42551be",
                ],
                ExternalIDs: nil,
                Fair: true,
                Name: "tcp-reset-rate-limiter",
                Unit: "pktps",
            },
            <*nbdb.Meter | 0xc00114d440>{
                UUID: "a554e5a6-ac3f-4c8d-a1d9-62c9baa77df2",
                Bands: [
                    "2e7bd3d6-1ba2-475e-af2e-db8da42551be",
                ],
                ExternalIDs: nil,
                Fair: true,
                Name: "arp-rate-limiter",
                Unit: "pktps",
            },
            <*nbdb.Meter | 0xc00114d4a0>{
                UUID: "ef364a77-499b-4c6a-9969-95b175cfa375",
                Bands: [
                    "2e7bd3d6-1ba2-475e-af2e-db8da42551be",
                ],
                ExternalIDs: nil,
                Fair: true,
                Name: "event-elb-rate-limiter",
                Unit: "pktps",
            },
            <*nbdb.Meter | 0xc00114d500>{
                UUID: "52e41877-485c-4b50-a734-d9b9c868e219",
                Bands: [
                    "2e7bd3d6-1ba2-475e-af2e-db8da42551be",
                ],
                ExternalIDs: nil,
                Fair: true,
                Name: "icmp4-error-rate-limiter",
                Unit: "pktps",
            },
            <*nbdb.Meter | 0xc00114d560>{
                UUID: "a7337e2d-5735-4c4c-b583-996f62afa80e",
                Bands: [
                    "2e7bd3d6-1ba2-475e-af2e-db8da42551be",
                ],
                ExternalIDs: nil,
                Fair: true,
                Name: "arp-resolve-rate-limiter",
                Unit: "pktps",
            },
            <*nbdb.Meter | 0xc00114d5c0>{
                UUID: "ff9eaf42-7480-40d1-be29-c2c6857f4001",
                Bands: [
                    "2e7bd3d6-1ba2-475e-af2e-db8da42551be",
                ],
                ExternalIDs: nil,
                Fair: true,
                Name: "icmp6-error-rate-limiter",
                Unit: "pktps",
            },
            <*nbdb.Copp | 0xc0023ed290>{
                UUID: "da82bb69-9efc-4d25-8792-0c3e058b04ca",
                ExternalIDs: nil,
                Meters: {
                    "reject": "reject-rate-limiter",
                    "icmp4-error": "icmp4-error-rate-limiter",
                    "svc-monitor": "svc-monitor-rate-limiter",
                    "event-elb": "event-elb-rate-limiter",
                    "icmp6-error": "icmp6-error-rate-limiter",
                    "arp": "arp-rate-limiter",
                    "tcp-reset": "tcp-reset-rate-limiter",
                    "bfd": "bfd-rate-limiter",
                    "arp-resolve": "arp-resolve-rate-limiter",
                },
                Name: "ovnkube-default",
            },
            <*nbdb.MeterBand | 0xc00070f080>{
                UUID: "2e7bd3d6-1ba2-475e-af2e-db8da42551be",
                Action: "drop",
                BurstSize: 0,
                ExternalIDs: nil,
                Rate: 25,
            },
        ]
```

IMHO it is simpler to just change the comment with this reasoning.